pypeit/core/procimg.py

```diff
@@ -392,28 +470,30 @@ def subtract_overscan(rawframe, datasec_img, oscansec_img,
         elif method.lower() == 'median':
             # Subtract scalar and continue
             no_overscan[data_slice] -= osfit
+            if var is not None:
+                _var[data_slice] += osvar
             continue
-        else:
-            raise ValueError('Unrecognized overscan subtraction method: {0}'.format(method))
 
         # Subtract along the appropriate axis
         no_overscan[data_slice] -= (ossub[:, None] if compress_axis == 1 else ossub[None, :])
+        if var is not None:
+            _var[data_slice] += (osvar[:,None] if compress_axis == 1 else osvar[None,:])
 
```

Comment:
I think this would be cleaner if you always just returned the noise on the overscan subtraction and then you can add that in quadrature to the variance outside of this routine if you so desire.  In that sense, the outputs are always the same and the inputs are immutable, which I think is a better coding model.

Reply:
Good call.  Just to clarify a couple of things: (1) the code needs the input variance to calculate the variance in the overscan subtraction, so I just have it return None as the 2nd object if no variance is provided, and (2) the code never altered the input `var`.  Regardless, I now have it just return the variance in the overscan estimate; i.e., the value that should be added to the "processing variance".  Considering the gymnastics I do in the calling function, it made good sense to just return the error contribution from the overscan subtraction instead of automatically adding it to the input variance.